### ISSUE TITLE
 TPC-H Query 4

### ISSUE THREAD

**Issue by alamb** [apache/arrow-datafusion#160]
*Note*: migrated from original JIRA: https://issues.apache.org/jira/browse/ARROW-11521

Intervals expressions not yet supported.

See https://github.com/apache/arrow/pull/9434

Error:

Error: NotImplemented("Unsupported Interval Expression with leading_field Some(Month)")

**Comment by andygrove** [comment]
Latest failures 

```
DateIntervalExpr does not support IntervalYearMonth
```

```
Error: NotImplemented("Physical plan does not support logical expression EXISTS (Subquery: Projection: #lineitem.l_orderkey, #lineitem.l_partkey, #lineitem.l_suppkey, #lineitem.l_linenumber, #lineitem.l_quantity, #lineitem.l_extendedprice, #lineitem.l_discount, #lineitem.l_tax, #lineitem.l_returnflag, #lineitem.l_linestatus, #lineitem.l_shipdate, #lineitem.l_commitdate, #lineitem.l_receiptdate, #lineitem.l_shipinstruct, #lineitem.l_shipmode, #lineitem.l_comment\n  Filter: #lineitem.l_orderkey = #orders.o_orderkey AND #lineitem.l_commitdate < #lineitem.l_receiptdate\n    TableScan: lineitem projection=None)")
t
```

**Comment by avantgardnerio** [comment]
Resolved by PR #2813 & PR #2797
